src/Behat/Context/UsersContext.php

```diff
@@ -39,33 +39,83 @@ public function userWithMailNotExists($mail) {
    *   True if the user should NOT have the specific roles.
    */
   public function userRoleCheck($role, $user = NULL, $not = FALSE) {
-    if (empty($user)) {
-      $current_user = $this->getUserManager()->getCurrentUser();
-      $account = $this->getCore()->loadUserByProperty('uid', $current_user->uid);
+    // Because Behat process is alive during the whole test suite, we should
+    // clear user cache.
+    // @TODO: We may clear only the user being checked. However, the process
+    // to get the uid is not direct. Let's just clear the complete cache for
+    // now.
+    $this->getCore()->staticEntityCacheClear('user');
+
+    $account = $this->locateAccount($user);
+
+    if (!$account) {
+      throw new \Exception("Given user does not exist!");
     }
-    else {
-      $property = $this->getCore()->getUserPropertyByName($user);
-      $account = $this->getCore()->loadUserByProperty($property, $user);
+
+    $roles_to_check = $this->roleString2Array($role);
+
+    // Get current roles for the user.
+    $account_roles = array_map('strtolower', $this->getCore()->getUserRoles($account));
+
+    // Calculate...
+    $common_roles = array_intersect($roles_to_check, $account_roles);
+    $roles_not_in_account = array_diff($roles_to_check, $account_roles);
+
+    // ...and check!
+    if (!$not && count($roles_not_in_account)) {
```

Comment:
This only will work if the user has exactly the same roles than the roles we want to check. If the user has one more role, it will throw the exception, but is a valid use case.

Reply:
```suggestion
    if (!$not && count($common_roles) == count($roles_not_in_account)) {
```

Reply:
```suggestion
    if (!$not && count($common_roles) == count($roles_to_check)) {
```

Reply:
It will work even in that case. Let's say we want to check roles a and b. And let's say the account has the roles a,b and c.

$diff is:
` [a,b] - [a, b, c] = []`

So count($diff) is 0.

The point is diff operation removes any item in the first array that is present in the second array, but if an item is present in the second array but not in the first no action is done.